### PR TITLE
[TINKERPOP-3076] Throw better error if request exceeds WriteBufferSize in Go

### PR DESCRIPTION
**Pull request by Cole-Greer**
https://issues.apache.org/jira/browse/TINKERPOP-3076

Gorilla has an implicit frame size limit imposed by WriteBufferSize. Requests which exceed this limit cannot be parsed by gremlin server and currently lead to cryptic errors in the server. This PR simply adds a client side check to throw a more descriptive error to the user. The resulting behaviour is very similar to how `maxContentLength` functions in the java driver.

VOTE +1

### CODE REVIEW

File: gremlin-go/driver/gorillaTransporter.go [hunk 1]
```diff
@@ -100,6 +101,9 @@ func (transporter *gorillaTransporter) Write(data []byte) error {
 			return err
 		}
 	}
+	if len(data) > transporter.connSettings.writeBufferSize {
+		return errors.New("request too large, consider increasing 'WriteBufferSize' configuration or split into smaller requests")
```
Comment (vkagamlyk) [inline gremlin-go/driver/gorillaTransporter.go:105]: all errors should be defined in `error_codes.go`
and used like `return newError(err0101ConnectionCloseError)`

**Comment by kenhuuu** [comment]
You should add an actual English error message to `en.json`

**Comment by vkagamlyk** [comment]
VOTE+1

**Comment by kenhuuu** [comment]
VOTE +1

**Comment by xiazcy** [comment]
VOTE +1

**Pull request merged**